packages/blast-core/lib/cudos-contract.js

```diff
@@ -19,18 +23,27 @@ module.exports.CudosContract = class CudosContract {
     this.#contractLabel = contractLabel
     this.#signer = signer
     this.#contractAddress = deployedContractAddress
-    this.#wasmPath = path.join(getProjectRootPath(), `artifacts/${contractLabel}.wasm`)
     this.#gasPrice = GasPrice.fromString(getGasPrice())
 
-    if (deployedContractAddress === null && !fs.existsSync(this.#wasmPath)) {
-      throw new BlastError(`Contract with label ${contractLabel} was not found, did you compile it?`)
+    if (deployedContractAddress === null) {
+      this.#wasmPath = path.join(getProjectRootPath(), `artifacts/${contractLabel}.wasm`)
+      if (!fs.existsSync(this.#wasmPath)) {
+        throw new BlastError(`Contract with label ${contractLabel} was not found, did you compile it?`)
+      }
     }
   }
 
-  async deploy(initMsg, signer = this.#signer, label = this.#contractLabel, funds) {
-    this.#signer = signer
+  async deploy(initMsg, options = {}) {
+    if (options.signer) {
+      this.#signer = options.signer
+    } else if (this.#contractAddress === null) {
```

Comment:
please try to remove the else if structure to become more clear, readable, and easy to maintan